app/services/teachers/manage.rb

```diff
@@ -13,71 +13,116 @@ def initialize(author:, teacher:, appropriate_body:)
     @appropriate_body = appropriate_body
   end
 
-  def update_name!(trs_first_name:, trs_last_name:)
-    @old_name = full_name
-    teacher.assign_attributes(trs_first_name:, trs_last_name:)
-    @new_name = full_name
-    record_name_change_event
-    teacher.save!
+  # Updates a teacher record with the provided attributes and creates appropriate events
+  def update_teacher!(attributes)
+    if teacher.new_record?
+      update_and_save_teacher(attributes)
+      record_teacher_creation_event
+    else
+      capture_old_values(attributes)
+      update_and_save_teacher(attributes)
+      record_change_events(attributes)
+    end
+
+    teacher
   end
 
-  def update_qts_awarded_on!(trs_qts_awarded_on:)
-    @old_award_date = teacher.trs_qts_awarded_on
-    teacher.assign_attributes(trs_qts_awarded_on:)
-    @new_award_date = teacher.trs_qts_awarded_on
-    record_award_change_event
+private
+
+  attr_reader :new_name, :old_name, :new_award_date, :old_award_date
+
+  # Updates the teacher with the provided attributes and saves the record
+  def update_and_save_teacher(attributes)
+    teacher.assign_attributes(attributes)
     teacher.save!
   end
 
-  def update_itt_provider_name!(trs_initial_teacher_training_provider_name:)
-    @itt_provider_before = teacher.trs_initial_teacher_training_provider_name
-    teacher.assign_attributes(trs_initial_teacher_training_provider_name:)
-    @itt_provider_after = teacher.trs_initial_teacher_training_provider_name
-    teacher.save!
+  # Captures the old values before updating the teacher record
+  def capture_old_values(attributes)
+    if name_attributes_changing?(attributes)
+      @old_name = current_full_name
+    end
+
+    if qts_award_date_changing?(attributes)
+      @old_award_date = teacher.trs_qts_awarded_on_was
+    end
   end
 
-private
+  # Records events for changes to the teacher record
+  def record_change_events(attributes)
+    if name_attributes_changing?(attributes)
+      @new_name = current_full_name
+      record_name_change_event
+    end
+
+    if qts_award_date_changing?(attributes)
+      @new_award_date = teacher.trs_qts_awarded_on
+      record_award_change_event
+    end
+  end
 
-  attr_reader :new_name, :old_name, :new_award_date, :old_award_date
+  # Records an event for a new teacher record
+  def record_teacher_creation_event
+    Events::Record.teacher_record_created!(
+      author:,
+      teacher:,
+      appropriate_body:,
+      trn: teacher.trn
```

Comment:
Could we avoid passing in `trn` by letting the record `heading` access `teacher.trn`